iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergSnapshotRefExec.java

```diff
@@ -56,16 +59,35 @@ public static void createBranch(Table table, AlterTableSnapshotRefSpec.CreateSna
         throw new IllegalArgumentException(String.format("Tag %s does not exist", tagName));
       }
     } else {
-      snapshotId = Optional.ofNullable(table.currentSnapshot()).map(snapshot -> snapshot.snapshotId()).orElse(null);
+      snapshotId = Optional.ofNullable(table.currentSnapshot()).map(Snapshot::snapshotId).orElse(null);
     }
-    ManageSnapshots manageSnapshots = table.manageSnapshots();
+    Transaction trans = null;
+    ManageSnapshots manageSnapshots;
     if (snapshotId != null) {
-      LOG.info("Creating a branch {} on an iceberg table {} with snapshotId {}", branchName, table.name(), snapshotId);
-      manageSnapshots.createBranch(branchName, snapshotId);
+      manageSnapshots = createBranch(table, isReplace, branchName, snapshotId);
     } else {
-      LOG.info("Creating a branch {} on an empty iceberg table {}", branchName, table.name());
-      manageSnapshots.createBranch(branchName);
+      if (isReplace) {
+        LOG.info("Replacing a branch {} on an empty iceberg table {}", branchName, table.name());
```

Comment:
> The empty iceberg table replace was bit tricky, I didn't find any other alt than dropping & recreating the branch under a transaction, will double check in the iceberg code, if there in any API which can do that magic, thought of truncate but it isn't replace, the same branch stays

I  checked Spark-iceberg code, and it will throw exception if replacing the branch on a empty table :
https://github.com/apache/iceberg/blob/dd2197f835aa3f180c0f49246055ce665f9fef59/spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/CreateOrReplaceBranchExec.scala#L62-L68

So i think we can do like spark. it's more simpler.
But if you think we need to support to replace branch on empty. I also have no objection. :)

Reply:
I don't have a usecase as such, it is just, If I call create branch on an empty iceberg table it will succeed but it won't when there is existing branch & some one uses replace, though he can himself achieve that by first dropping the branch & then calling create again.

if there are even some corner cases where we can be screwed by the current impl, we should definitely fallback to throwing exception.

Let me know, what you think, I will update the PR accordingly :-)

Reply:
ok,Let us follow spark-iceberg behavior, just throw exception on empty table when replacing branch.

Reply:
I have updated the PR to throw exception in this case, added a test as well to validate the case